From 3473ea3488c6d41e09b41c53f8f433b78156cc4c Mon Sep 17 00:00:00 2001
From: Dave Kover <dkover@cyngn.com>
Date: Fri, 9 Dec 2016 10:47:17 -0700
Subject: [PATCH 13/37] Themes: Expose resolver hardcoded colors

commit dbbd5e70cc65002df41561474b03362022dd6716
Author: Dave Kover <dkover@cyngn.com>
Date:   Wed Feb 18 16:11:14 2015 -0800

    Themes: Expose resolver hardcoded colors

    Expose background colors of the resolver list.

    Change-Id: I3a0a460c5ffe0f5057b3b9ec92faa7a3e09c9e01

commit 0343eb126f3901a3857791137f74fa805bb9d75c
Author: Thyrus11 <thyrus11@gmail.com>
Date:   Sat Feb 21 07:19:42 2015 +0100

    Themes: Make resolver list fully themeable

    Follow-up on commit cc9e3b8fcba95b911d1cda36f7770c410058aa8b.

    Change-Id: I3f006a1157db9d0b151a4fe8edf50e7edc7a0b9f

commit c7d973809488b801e8c708d740009f1233bb762e
Author: Nicholas Chum <nicholaschum@gmail.com>
Date:   Sun Nov 8 05:27:28 2015 -0500

    Themes: Allow Resolver List BG to be fully themed

    We are able to trace the activity of the new resolver/chooser through
    different
    methods, thus leading us to the Java file:
    \com\android\internal\app\ChooserActivity.java
    Here we see that the exposed "chooser_service_row_background_color" is
    available, but not the rest of the activity, so we look into R.layout's,
    and we
    find chooser_grid to be the only one containing hardcoded
    "@color/white" values (as this is framework, we assume this is also
    known as
    "android:color/white" to themers).

    Expose all "@color/white" values from this file to resolver_list_bg.

    Change-Id: I286d92b5d1f672c8adb3c0af1951793521536d90

Change-Id: Iec7951147bbbc99aee6b06ae50c1acc7b9c01a7f
Signed-off-by: Adarsh-MR <adarshmr1998@gmail.com>
---
 core/res/res/layout/chooser_grid.xml                   | 6 +++---
 core/res/res/layout/resolver_different_item_header.xml | 2 +-
 core/res/res/layout/resolver_list.xml                  | 8 ++++----
 core/res/res/layout/resolver_list_with_default.xml     | 6 +++---
 core/res/res/values/projekt_colors.xml                 | 3 +++
 5 files changed, 14 insertions(+), 11 deletions(-)

diff --git a/core/res/res/layout/chooser_grid.xml b/core/res/res/layout/chooser_grid.xml
index 7dee2af3abb..13bc8129d54 100644
--- a/core/res/res/layout/chooser_grid.xml
+++ b/core/res/res/layout/chooser_grid.xml
@@ -31,7 +31,7 @@
             android:layout_alwaysShow="true"
             android:elevation="8dp"
             android:paddingStart="16dp"
-            android:background="@color/white" >
+            android:background="@color/resolver_list_bg" >
         <TextView android:id="@+id/profile_button"
                   android:layout_width="wrap_content"
                   android:layout_height="48dp"
@@ -73,7 +73,7 @@
             android:id="@+id/resolver_list"
             android:clipToPadding="false"
             android:scrollbarStyle="outsideOverlay"
-            android:background="@color/white"
+            android:background="@color/resolver_list_bg"
             android:elevation="8dp"
             android:listSelector="@color/transparent"
             android:divider="@null"
@@ -84,7 +84,7 @@
               android:layout_width="match_parent"
               android:layout_height="wrap_content"
               android:layout_alwaysShow="true"
-              android:background="@color/white"
+              android:background="@color/resolver_list_bg"
               android:text="@string/noApplications"
               android:padding="32dp"
               android:gravity="center"
diff --git a/core/res/res/layout/resolver_different_item_header.xml b/core/res/res/layout/resolver_different_item_header.xml
index 58891363e20..201c8c686fe 100644
--- a/core/res/res/layout/resolver_different_item_header.xml
+++ b/core/res/res/layout/resolver_different_item_header.xml
@@ -29,6 +29,6 @@
     android:paddingEnd="16dp"
     android:paddingTop="8dp"
     android:paddingBottom="8dp"
-    android:background="@color/white"
+    android:background="@color/resolver_list_bg"
     android:elevation="8dp"
     />
diff --git a/core/res/res/layout/resolver_list.xml b/core/res/res/layout/resolver_list.xml
index 40c99419613..92540e19d18 100644
--- a/core/res/res/layout/resolver_list.xml
+++ b/core/res/res/layout/resolver_list.xml
@@ -30,7 +30,7 @@
         android:layout_height="wrap_content"
         android:layout_alwaysShow="true"
         android:elevation="8dp"
-        android:background="@color/white">
+        android:background="@color/resolver_list_bg">
 
         <TextView
             android:id="@+id/profile_button"
@@ -69,7 +69,7 @@
         android:id="@+id/resolver_list"
         android:clipToPadding="false"
         android:scrollbarStyle="outsideOverlay"
-        android:background="@color/white"
+        android:background="@color/resolver_list_bg"
         android:elevation="8dp"
         android:nestedScrollingEnabled="true"
         android:scrollIndicators="top|bottom"
@@ -78,7 +78,7 @@
     <TextView android:id="@+id/empty"
               android:layout_width="match_parent"
               android:layout_height="wrap_content"
-              android:background="@color/white"
+              android:background="@color/resolver_list_bg"
               android:elevation="8dp"
               android:layout_alwaysShow="true"
               android:text="@string/noApplications"
@@ -99,7 +99,7 @@
         android:orientation="horizontal"
         android:layoutDirection="locale"
         android:measureWithLargestChild="true"
-        android:background="@color/white"
+        android:background="@color/resolver_list_bg"
         android:paddingTop="8dp"
         android:paddingBottom="8dp"
         android:paddingStart="12dp"
diff --git a/core/res/res/layout/resolver_list_with_default.xml b/core/res/res/layout/resolver_list_with_default.xml
index 8101183d4e2..502d615cae7 100644
--- a/core/res/res/layout/resolver_list_with_default.xml
+++ b/core/res/res/layout/resolver_list_with_default.xml
@@ -29,7 +29,7 @@
         android:layout_height="wrap_content"
         android:layout_alwaysShow="true"
         android:orientation="vertical"
-        android:background="@color/white"
+        android:background="@color/resolver_list_bg"
         android:elevation="8dp">
 
         <LinearLayout
@@ -111,7 +111,7 @@
             android:paddingBottom="8dp"
             android:paddingStart="12dp"
             android:paddingEnd="12dp"
-            android:background="@color/white"
+            android:background="@color/resolver_list_bg"
             android:elevation="8dp">
 
             <Button
@@ -151,7 +151,7 @@
         android:id="@+id/resolver_list"
         android:clipToPadding="false"
         android:scrollbarStyle="outsideOverlay"
-        android:background="@color/white"
+        android:background="@color/resolver_list_bg"
         android:elevation="8dp"
         android:nestedScrollingEnabled="true"
         android:divider="@null" />
diff --git a/core/res/res/values/projekt_colors.xml b/core/res/res/values/projekt_colors.xml
index c2355ffd594..c16b75d4369 100644
--- a/core/res/res/values/projekt_colors.xml
+++ b/core/res/res/values/projekt_colors.xml
@@ -28,4 +28,7 @@
     <!-- Toast text and bg colors -->
     <color name="toast_bg_color">#E6EEEEEE</color>
     <color name="toast_text_color">@color/primary_text_default_material_light</color>
+
+    <!-- Resolver -->
+    <color name="resolver_list_bg">@color/white</color>
 </resources>
-- 
2.18.0

